### DISCUSSION TITLE
Verifying apt installation

### DISCUSSION THREAD

**Discussion by alexjaw** [gitbls/sdm#47] [Q&A] [answered]
Wonder what you do to ensure that apt installs have been successful?

So, I have quite a large number of packets to install and they (of course) have to be correctly installed for subsequent operations. I wonder how to ensure that installation has been successful in an automatic manner. One possibility is to scan through the apt log produced by sdm. Another possibility is to use dpkg, for example: dpkg -s <package> | grep Status:

Any comments or suggestions?

Cheers
Alex

**Comment by gitbls** [comment]
I typically review the contents of /etc/sdm/apt.log and /etc/sdm/history each time I do a customization. Partly because, as the maintainer, I need to be on the lookout for problems, but mostly because I like to make sure that things actually worked as I expected. It's much quicker/cheaper to check before spending the time to burn and boot the system, only to discover that something didn't work correctly.

If you're having apt install problems, one possible help is to use the `--debug apt` switch. This causes sdm to do the app/xapp installs one package at a time, instead of feeding the whole list to apt all at once.

As far as checking logs in an automatic manner, I think it's pretty much scanning the above 2 files. All errors in `history` have a `?` in them, and all warnings have a `%`. And, the apt errorrs have the apt `E:` in them in `apt.log`.

In looking at the sdm apt code I see that there is no sdm messaging of apt errors done into `history`. I'm going to look into improving that so that the occurence of an apt failure is noted in `history`.

**Comment by gitbls** [comment, marked as answer]
The change to note the occurrence of apt failures is implemented in V7.5
